web/frontend/src/pages/election/components/SubjectComponent.tsx

```diff
@@ -0,0 +1,271 @@
+      return false;
+    };
+
+    const findRanks = (id: string) => {
+      if (Ranks) {
+        if (Ranks.find((rank) => rank.ID === id)) {
+          const rankFound = Ranks.find((rank) => rank.ID === id);
+          return (
+            <QuestionComponent
+              key={`rank${rankFound.ID}`}
+              obj={rankFound}
+              notifyParent={notifySubject(rankFound.ID, 'RANK')}
+              removeQuestion={removeChildQuestion(rankFound.ID, 'RANK')}
+              type={'RANK'}
+            />
+          );
+        }
+        return false;
+      }
+      return false;
+    };
+
+    const findTexts = (id: string) => {
+      if (Texts) {
+        if (Texts.find((text) => text.ID === id)) {
+          const textFound = Texts.find((rank) => rank.ID === id);
+          return (
+            <QuestionComponent
+              key={`text${textFound.ID}`}
+              obj={textFound}
+              notifyParent={notifySubject(textFound.ID, 'TEXT')}
+              removeQuestion={removeChildQuestion(textFound.ID, 'TEXT')}
+              type={'TEXT'}
+            />
+          );
+        }
+        return false;
+      }
+      return false;
+    };
+
+    const findSelects = (id: string) => {
+      if (Selects) {
+        if (Selects.find((select) => select.ID === id)) {
+          const selectFound = Selects.find((rank) => rank.ID === id);
+          return (
+            <QuestionComponent
+              key={`select${selectFound.ID}`}
+              obj={selectFound}
+              notifyParent={notifySubject(selectFound.ID, 'SELECT')}
+              removeQuestion={removeChildQuestion(selectFound.ID, 'SELECT')}
+              type={'SELECT'}
+            />
+          );
+        }
+        return false;
+      }
+      return false;
+    };
+
+    const orderData = () => {
+      setComponents(
+        Order.map((id) => {
+          if (findSubject(id)) {
+            return findSubject(id);
+          } else if (findRanks(id)) {
+            return findRanks(id);
+          } else if (findTexts(id)) {
+            return findTexts(id);
+          } else if (findSelects(id)) {
+            return findSelects(id);
+          } else return null;
+        })
+      );
+    };
+
+    orderData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [Title, Subjects, Ranks, Selects, Texts, Order, nestedLevel]);
+
+  return (
+    <div className="ml-4 mb-4 mr-2 shadow-lg rounded-md">
+      <DeleteButton text="Subject" onClick={removeSubject} />
+      <input
+        value={Title}
+        onChange={(e) => setSubject({ ...subject, Title: e.target.value })}
+        name="Title"
+        type="text"
+        placeholder="Enter the Subject Title"
+        className="ml-2 mt-2 mb-2 border rounded-md text-md w-60"
+      />
+      <div className="ml-2">ID: {subject.ID}</div>
+      {components.map((component) => component)}
+
+      <div className="flex hidden justify-between overflow-x-auto sm:flex sm:pr-2">
```

Comment:
What is this hidden section for ? Is it for the mobile version ?

Reply:
In the mobile version, this div won't be visible (with all the add ranks, selects, texts and subject buttons). So we have a more personalized view in the mobile version compared to the laptop one.